xwiki-commons-core/xwiki-commons-job/xwiki-commons-job-default/src/main/java/org/xwiki/job/internal/DefaultJobStatusStore.java

```diff
@@ -211,13 +209,11 @@ private String encode(String name)
         String encoded;
 
         if (name != null) {
-            try {
-                encoded = URLEncoder.encode(name, DEFAULT_ENCODING);
-            } catch (UnsupportedEncodingException e) {
-                // Should never happen
-
-                encoded = name;
-            }
+            encoded = URLEncoder.encode(name, StandardCharsets.UTF_8)
+                // Replace characters that might be problematic in file systems.
+                .replace("+", "%20")
```

Comment:
Isn't URLEncoder already supposed to encode the `+` characters as "%2B" ?

Reply:
Yes, but it encodes spaces as `+`.

Reply:
Right, not a very accurate encoding...

Reply:
In the end, it's probably fine to leave the `+` characters as-is, it's just not a very good encoding (and doesn't work in URLs in some cases, see https://jira.xwiki.org/browse/XWIKI-22956).

Reply:
The code is using `FileSystemStoreUtils#encode` which keeps `+` as encoding of spaces.